### ISSUE TITLE
Incorrect bytecode versions

### ISSUE THREAD

**Issue by codefromthecrypt** [openzipkin/zipkin-reporter-java#235]
## Describe the Bug

Using brave, I noticed the bytecode  versions aren't assigned correctly. For example, the URLConnectionSender should be compiled java 6 and it isn't.

```
 backend         | org.springframework.beans.factory.CannotLoadBeanClassException: Error loading class [zipkin2.reporter.beans.URLConnectionSenderFactoryBean] for bean with name 'sender' defined in ServletContext resource [/WEB-INF/applicationContext.xml]: problem with class file or dependent class; nested exception is java.lang.UnsupportedClassVersionError: zipkin2/reporter/beans/URLConnectionSenderFactoryBean : Unsupported major.minor version 52.0
```

## Steps to Reproduce

look here https://github.com/openzipkin/brave-example/pull/97

## Expected Behaviour

We probably need to lift infrastructure from zipkin to correct this.

**Comment by codefromthecrypt** [comment]
double-checked files inside zipkin's released 2.26.0 jar are correct, so the problem is here, not there

```bash
$ file zipkin2/CheckResult.class
zipkin2/CheckResult.class: compiled Java class data, version 50.0 (Java 1.6)
```

**Comment by codefromthecrypt** [comment]
brave classes are busted also, so will follow-up there, too. This happened when we switched off retrolambda.

```bash
$ file brave/CurrentSpanCustomizer.class
brave/CurrentSpanCustomizer.class: compiled Java class data, version 52.0 (Java 1.8)
```
